asapdiscovery-alchemy/asapdiscovery/alchemy/schema/fec.py

```diff
@@ -60,6 +61,123 @@ def to_solvent_component(self) -> gufe.SolventComponent:
         return gufe.SolventComponent(**self.dict(exclude={"type"}))
 
 
+class AdaptiveSettings(_SchemaBase):
+    """
+    A settings class to encode settings for adaptive settings. These were recommended by OpenFE.
+    """
+
+    type: Literal["AdaptiveSettings"] = "AdaptiveSettings"
+    adaptive_sampling: bool = Field(
+        True,
+        description="If True, will enable increase in production length of simulations given a `adaptive_sampling_multiplier` and `adaptive_sampling_threshold`.",
+    )
+    adaptive_sampling_multiplier: float = Field(
+        2.0,
+        description="The number of times more production simulation length (sampling time) that will be assigned to edges whose mapping scoring falls below the `adaptive_sampling_threshold`.",
+    )
+    adaptive_sampling_threshold: float = Field(
+        0.5,
+        description="The threshold that separates edges that are expected to perform well (higher; regular production simulation time) and poorly (lower; regular production simulation time * `adaptive_sampling_multiplier`). Recommended settings are 0.5 (LOMAP scorer) or 0.85 (PERSES scorer).",
+    )
+    adaptive_solvent_padding: bool = Field(
+        True,
+        description="Whether or not to use adaptive solvent padding; typically the complex phase can handle smaller padding size.",
+    )
+    solvent_padding_complex: FloatQuantity["nanometer"] = Field(  # noqa: F821
+        1.0 * OFFUnit.nanometer,
+        description="The solvent padding (in nm) to use for the complex phase of each edge.",
+    )
+    solvent_padding_solvated: FloatQuantity["nanometer"] = Field(  # noqa: F821
+        1.2 * OFFUnit.nanometer,
+        description="The solvent padding (in nm) to use for the solvated phase of each edge.",
+    )
+
+    def get_adapted_sampling_protocol(
+        self,
+        scorer_method: str,
+        mapping: "LigandAtomMapping",
+        protocol: openfe.protocols.openmm_rfe.RelativeHybridTopologyProtocol,
+        base_sampling_length: FloatQuantity["nanometer"],  # noqa: F821
+    ) -> openfe.protocols.openmm_rfe.RelativeHybridTopologyProtocol:
+        """
+        It's advisable to increase simulation time on edges that are expected to be less reliable. There
+        Aren't many good estimators for this, but the network planner edge scoring is a decent approximation.
+
+        If the edge scoring (computed using `scorer_method`) is below the `adaptive_sampling_threshold` the
+        simulation time is multiplied by `adaptive_sampling_multiplier`. Just to be sure, we use the base
+        protocol's sampling time and not the provided edge protocol sampling time as a base value.
+
+        Returns the adjusted OpenFE Protocol.
+        """
+        if scorer_method == "default_lomap":
+            scorer = lomap_scorers.default_lomap_score
+        elif scorer_method == "default_perses":
+            scorer = perses_scorers.default_perses_scorer
+        else:
+            raise ValueError(
```

Comment:
If this is on by default, then any scoring method other than these two will cause an exception. Are we sure that is what we want? We could just return protocol with a warning, or is there anything else stopping us from using other scorers?

Reply:
it's a fair point, this check might not be needed at all because we're probably checking for this upstream. There aren't any scorers available in OpenFE currently, but removing the check here would make implementing any upcoming scorers less painful.

Reply:
there are only two scorers available in OpenFE - any new scorers would need to be added here. I'll have to think about whether it's worth automating this step, also because I've heard through the grapevine that OpenFE will be porting mapping scorings directly through their transformations soon anyway.

Reply:
@jthorton thoughts?

Reply:
I would do this for now and then remove this once the LigandNetworks correctly stores the edge score which should be [fixed soon](https://github.com/OpenFreeEnergy/gufe/pull/369) as you will always want to use the same scorer which planned the network.